### ISSUE TITLE
Yolov 13 has been released

### ISSUE THREAD

**Issue by lix8886** [ultralytics/ultralytics#21181]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

https://github.com/iMoonLab/yolov13

啥时候官方库能够合并兼容一下？
When will the official libraries be merged for yolov13?

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lix8886, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are already answered there.

This is an automated response to help guide your issue, and an Ultralytics engineer will also assist you soon 😊.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

For feature requests or compatibility questions, please include as much relevant information as possible, such as links, motivation, and potential use cases.

Join the Ultralytics community where it suits you best! For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by yang-fei** [comment]
Yolov13 is built based on ultralytics. Your code should be easy to add it

**Comment by Mariuscheng** [comment]
Is this support cuda 12.8?

**Comment by glenn-jocher** [comment]
Thanks for the interest! Just to clarify, the YOLOv13 repository you've linked is a third-party implementation, not an official Ultralytics model. Our latest official model is [YOLO11](https://docs.ultralytics.com/models/yolo11/) which delivers state-of-the-art performance across multiple tasks.

@Mariuscheng the CUDA compatibility question would need to be directed to the iMoonLab repository since that's their implementation. For official Ultralytics models, we support the standard PyTorch CUDA requirements.

We focus our development efforts on our official model line, so there are no current plans to integrate third-party YOLO variants. I'd recommend trying YOLO11 which offers excellent performance and is actively maintained with the full Ultralytics ecosystem support.

**Comment by zshar7** [comment]
Does this mean that YOLOv13 would not be implemented into the library like YOLOv12 and v10 which are non-Ultralytics developed?

I was thinking there may be an update so if you put the model as YOLOv13 like:
```
model = YOLO("yolo13n.pt")
```
It would automatically download the pretrained weights instead of having to manually download and reference.
I'm happy that you can do this for models like YOLOv12 but concerned that your comment implies that this practice is discontinued and will not be done with YOLOv13.

**Comment by glenn-jocher** [comment]
@zshar7 You raise a good point! The practice isn't discontinued - we do evaluate and integrate promising third-party YOLO variants when they offer significant value to the community. YOLOv10 and YOLOv12 were integrated because they represented meaningful advances with good compatibility.

Each integration decision depends on factors like performance gains, architectural innovation, community demand, and maintenance feasibility. We're always open to evaluating new models, but there's no automatic integration for every variant that appears.

For YOLOv13 specifically, we'd need to assess whether it provides compelling advantages over our current offerings, especially given that YOLO11 already delivers state-of-the-art performance. If you're interested in this particular model's integration, I'd encourage opening a dedicated feature request with performance comparisons and use case details.

**Comment by zshar7** [comment]
I'd be very willing to help with a PR on YOLOv13 being added to Ultralytics.
I was very hesitant thinking there is some barrier to add YOLOv13 (copyright?) or someone is currently working on a PR.

This would also be my first PR to this repo.

**Comment by glenn-jocher** [comment]
@zshar7 That's fantastic - we always appreciate community contributions! There are no copyright barriers, but before jumping into implementation, I'd recommend starting with that dedicated feature request I mentioned. This helps ensure alignment with the project's direction and gets community feedback first.

The process would be: 1) Create a detailed feature request with performance benchmarks comparing YOLOv13 to YOLO11/other models, 2) Get community discussion and maintainer approval, 3) Then proceed with the PR implementation. Check out our [contributing guidelines](https://docs.ultralytics.com/help/contributing/) for the full process - they'll help make your first PR smooth!

**Comment by kennedy-kitoko** [comment]
📊 YOLOv13 Performance Validation - Agricultural Applications Study
Hello @glenn-jocher and Ultralytics team! 👋
Following the discussion on YOLOv13 integration (#21181), I'd like to contribute to the evaluation process with empirical performance benchmarks that directly address the technical requirements mentioned.
🔬 Comprehensive Comparative Study Completed
I've conducted a rigorous empirical study comparing YOLOv12 vs YOLOv13 architectures with SDPA and Flash Attention mechanisms, specifically validated on agricultural object detection tasks.
📈 Key Performance Results:

YOLOv13 + SDPA: 82.9% mAP50, 73.5% recall
YOLOv13 + Flash: 82.3% mAP50, 89.4% precision
YOLOv12 + SDPA: 76.7% mAP50, 66.4% recall
Performance Gain: +6.2% mAP50 improvement (YOLOv13 vs YOLOv12)

🌾 Agricultural Domain Validation

Dataset: Weeds-3 agricultural object detection (3,664 training images)
Hardware: AMD Ryzen 9 7945HX + RTX 4060 (standardized environment)
Methodology: 4 configurations tested, 20-100 epochs training
Innovation Focus: HyperACE architecture + attention mechanisms optimization

🎯 Study Repository & Documentation
📂 Complete Study: https://github.com/kennedy-kitoko/yolov12-vs-yolov13-attention-study
The repository includes:

Comprehensive experimental methodology
Detailed performance analysis and visualizations
Hardware specifications and reproducibility guidelines
Agricultural-specific optimization insights
Training curves, validation metrics, and comparative analysis

🏆 Community Recognition
This work has been publicly recognized by the official YOLO community:

Official Ultralytics Twitter account responded positively
Muhammad Rizwan Munawar (YOLOv11 contributor & Ultralytics growth engineer) praised the innovation: "Wow, very nice 🎉"

💡 Contribution to Integration Decision
These empirical benchmarks provide the "performance comparisons and use case details" requested for YOLOv13 evaluation. The study demonstrates:
✅ Significant Performance Gains: +6.2% mAP50 validated
✅ Real-world Application: Agricultural object detection proven
✅ Architectural Innovation: HyperACE + attention mechanisms validated
✅ Community Value: Specialized domain expertise contributing to ecosystem
🤝 Collaboration Offer
As mentioned by @zshar7, I'm willing to contribute to the YOLOv13 integration effort. My empirical study provides the technical foundation needed for informed decision-making regarding YOLOv13's value proposition within the Ultralytics ecosystem.
Ready to assist with:

Technical documentation and benchmarking
Agricultural domain expertise and validation
Integration testing and optimization
Community education and adoption support

📧 Contact & Institution
Kennedy Kitoko (马恒)
Beijing Institute of Technology
Email: [kitokokennedy13@gmail.com](mailto:kitokokennedy13@gmail.com)
Looking forward to contributing to the continued excellence of the Ultralytics ecosystem! 🚀
P.S.: This study was completed within days of YOLOv13's release (June 26-27, 2025), demonstrating rapid technical adaptation and validation capabilities that could benefit future model evaluations.

Feel free to review the complete study and let me know if additional benchmarks or specific metrics would be helpful for the integration evaluation process.

**Comment by zshar7** [comment]
@kennedy-kitoko This is exciting news to find regarding YOLOv13.
I will take some time to look through your research but at first glance it is showing promising results.
Do want to kindly note that there is a dedicated discussion about YOLOv13 integration in Issue #21243. Would be nice to copy and paste your message there for documentation.
YOLOv13 has been having issues as it has been seen that the data it shows about YOLO12 was different than the accepted metrics and there was no presence of concrete mAP50-95 values. I reached out to the maintainers of the GitHub repository, and received little and not sufficient answers.
I believe to pave the way or fully reject YOLOv13's integration into Ultralytics is to do a third party study to compare YOLO11, YOLO12, and YOLOv13 and show if YOLOv13 yields significant results.
Your results are one of the few we might need. Having COCO be a dataset of testing is crucial (in my opinion).
There was a third party article claiming YOLOv13 is really good however there are many discrepancies (https://so-development.org/comparing-yolov12-and-yolov13-the-evolution-of-real-time-object-detection/)
With that being said, if you are willing to collaborate on a third party project evaluating these models let me know. I haven't been able to make progress on my end because of some other projects I have but I'm ready to start!

**Comment by kennedy-kitoko** [comment]
🔬 Independent YOLOv13 Validation - Addressing Performance Discrepancies
@zshar7 @glenn-jocher @Y-T-G Thank you for highlighting these critical performance discrepancies. This is exactly why rigorous third-party validation is essential.
🎯 Addressing the Core Issues
My empirical study provides concrete data that addresses the specific concerns raised:
🚨 Issue Identified: Conflicting mAP scores between YOLOv13 paper and official YOLO12 benchmarks
✅ My Solution: Independent validation with standardized methodology and reproducible results
📊 Concrete Metrics (Not "XX% increases")
Agricultural Dataset (Weeds-3) Results:

YOLOv13 + SDPA: 82.9% mAP50, 73.5% recall
YOLOv12 + SDPA: 76.7% mAP50, 66.4% recall
Net Performance: +6.2% mAP50 improvement (concrete numbers)

Hardware Standardized: AMD Ryzen 9 7945HX + RTX 4060 (fully reproducible)
🔬 Proposed Solution: Comprehensive COCO Validation
To resolve the performance discrepancies mentioned by @Y-T-G, I propose conducting a definitive third-party study:
📋 Validation Protocol:

COCO Dataset (industry standard benchmark)
Standardized Hardware (documented specifications)
Identical Training Protocols (reproducible methodology)
Multiple Runs (statistical significance)
Complete Transparency (open data, open methodology)

💡 Value Proposition for Ultralytics
This study will provide the "clear, reproducible improvements" that @glenn-jocher specified as integration requirements:
✅ Resolve Metric Discrepancies - Independent validation of conflicting claims
✅ Speed-Accuracy Trade-off - Comprehensive latency vs performance analysis
✅ Training Stability - Multi-run validation and convergence analysis
✅ Broad Compatibility - Cross-platform deployment testing
🎯 Study Scope & Timeline
Phase 1 : COCO training and validation
Phase 2: Performance analysis and verification
Phase 3 : Community report and recommendations
🤝 Collaboration with @zshar7
As @zshar7 mentioned independent research needs, I'm ready to collaborate on this critical validation. My infrastructure is prepared and methodology proven.
🏆 Expected Outcomes
This study will provide Ultralytics with the definitive data needed for informed decision-making:

If YOLOv13 shows clear gains: Evidence-based integration recommendation
If YOLOv13 shows regression: Clear rejection with documented reasoning
Either way: Community gets honest, transparent validation

📊 Repository & Methodology
Study Repository: https://github.com/kennedy-kitoko/yolov12-vs-yolov13-attention-study
Institution: Beijing Institute of Technology
Contact: [kitokokennedy13@gmail.com](mailto:kitokokennedy13@gmail.com)
💪 Commitment to Scientific Integrity
I'm committed to honest evaluation - if YOLOv13 doesn't meet Ultralytics standards, my study will document that clearly. The goal is truth, not advocacy.
Ready to start the COCO validation immediately and provide the community with the concrete, reproducible data needed to resolve these discrepancies.

**Comment by glenn-jocher** [comment]
@kennedy-kitoko @zshar7 Thanks for the detailed research and willingness to collaborate! Your agricultural validation is interesting, but for integration consideration we'd need benchmarks on standard datasets like COCO using the same evaluation protocol as our existing models.

As mentioned earlier, please start with a dedicated feature request that includes these comparative benchmarks against YOLO11 (our current SOTA). The [contributing guidelines](https://docs.ultralytics.com/help/contributing/) outline the proper process - feature request first, then community discussion, then potential PR if approved.

COCO validation comparing YOLOv13 vs YOLO11 using identical training protocols would provide the most actionable data for an integration decision. Looking forward to seeing those results in a formal feature request!

**Comment by zshar7** [comment]
@glenn-jocher I'm shifting this discussion to #21243 because this follows the contribution guidelines 😀

**Comment by glenn-jocher** [comment]
@zshar7 Perfect! That's exactly the right approach - moving to #21243 ensures proper organization and follows our contribution process. Looking forward to seeing the continued discussion and any COCO benchmarks there.
